includes/templates/class-amp-template.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+class AMP_Template {
+	const SITE_ICON_SIZE = 32;
+	const CONTENT_MAX_WIDTH = 600;
+
+	// Needed for 0.3 back-compat
+	const DEFAULT_NAVBAR_BACKGROUND = '#0a89c0';
+	const DEFAULT_NAVBAR_COLOR = '#fff';
+	
+	protected $content_max_width;
+	protected $template_dir;
+	protected $amp_base_template_data;
+	protected $data;
+
+	public function __construct() {
+		$this->content_max_width = self::CONTENT_MAX_WIDTH;
+		if ( isset( $GLOBALS['content_width'] ) && $GLOBALS['content_width'] > 0 ) {
+			$content_max_width = $GLOBALS['content_width'];
+		}
+		$this->content_max_width = apply_filters( 'amp_content_max_width', $content_max_width );
+
+		// Keep two filter hooks for backward compatibility (prefer the more general second option)
+		$this->template_dir = apply_filters( 'amp_post_template_dir', AMP__DIR__ . '/theme-templates' );
+		$this->template_dir = apply_filters( 'amp_template_dir', AMP__DIR__ . '/theme-templates' );
+
+		$this->init_amp_template_data();
+	}
+
+	private function init_amp_template_data() {
+		$this->amp_base_template_data = array(
+			'document_title' => function_exists( 'wp_get_document_title' ) ? wp_get_document_title() : wp_title( '', false ), // back-compat with 4.3
+			'html_tag_attributes' => array(),
+			'site_icon_url' => apply_filters( 'amp_site_icon_url', function_exists( 'get_site_icon_url' ) ? get_site_icon_url( self::SITE_ICON_SIZE ) : '' ),
+			'placeholder_image_url' => amp_get_asset_url( 'images/placeholder-icon.png' ),
+			'content_max_width' => $this->content_max_width,
+			'amp_runtime_script' => 'https://cdn.ampproject.org/v0.js',
+			'amp_styles' => array(),
+			'amp_component_scripts' => array(),
+			'font_urls' => array(
+				'merriweather' => 'https://fonts.googleapis.com/css?family=Merriweather:400,400italic,700,700italic',
+			),
+			'customizer_settings' => array(),
+			'home_url' => home_url(),
+			'blog_name' => get_bloginfo( 'name' ),
+			'body_class' => '',
+			/**
+			 * Add amp-analytics tags.
+			 *
+			 * This filter allows you to easily insert any amp-analytics tags without needing much heavy lifting.
+			 *
+			 * @since 0.4
+			 *.
+			 * @param	array	$analytics	An associative array of the analytics entries we want to output. Each array entry must have a unique key, and the value should be an array with the following keys: `type`, `attributes`, `script_data`. See readme for more details.
+			 * @param	object	$post	The current post.
+			 */
+			'amp_analytics' => apply_filters( 'amp_post_template_analytics', array(), $this->post ),
```

Comment:
Do we want to rename this filter as well (with proper back-compat handling)? One issue is that because of load order, `$this->post` will be null (`AMP_Template::__construct` is called before `$this->post` is assigned in `AMP_Post_Template`.

Maybe we change this filter to `amp_template_analytics` and remove the post object and move this into the `AMP_Post_Template` constructor where we initialize the other post data?